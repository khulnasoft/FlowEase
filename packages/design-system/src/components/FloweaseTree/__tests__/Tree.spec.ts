import { render } from '@testing-library/vue';
import FloweaseTree from '../Tree.vue';

describe('components', () => {
	describe('FloweaseTree', () => {
		it('should render simple tree', () => {
			const wrapper = render(FloweaseTree, {
				props: {
					value: {
						hello: 'world',
					},
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});

		it('should render tree', () => {
			const wrapper = render(FloweaseTree, {
				props: {
					value: {
						hello: {
							test: 'world',
						},
						options: ['yes', 'no'],
					},
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});

		it('should render tree with slots', () => {
			const wrapper = render(FloweaseTree, {
				props: {
					value: {
						hello: {
							test: 'world',
						},
						options: ['yes', 'no'],
					},
				},
				slots: {
					label: '<span>label</span>',
					value: '<span>value</span>',
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});

		it('should render each tree with node class', () => {
			const wrapper = render(FloweaseTree, {
				props: {
					value: {
						hello: {
							test: 'world',
						},
						options: ['yes', 'no'],
					},
					nodeClass: 'nodeClass',
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});
	});
});
