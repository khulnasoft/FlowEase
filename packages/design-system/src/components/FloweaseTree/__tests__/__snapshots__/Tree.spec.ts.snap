// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components > FloweaseTree > should render each tree with node class 1`] = `
"<div class="flowease-tree">
  <div class="nodeClass">
    <div><span>hello</span>
      <div class="flowease-tree">
        <div class="nodeClass indent">
          <div class="simple"><span>test</span><span>:</span><span>world</span></div>
        </div>
      </div>
    </div>
  </div>
  <div class="nodeClass">
    <div><span>options</span>
      <div class="flowease-tree">
        <div class="nodeClass indent">
          <div class="simple"><span>0</span><span>:</span><span>yes</span></div>
        </div>
        <div class="nodeClass indent">
          <div class="simple"><span>1</span><span>:</span><span>no</span></div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`components > FloweaseTree > should render simple tree 1`] = `
"<div class="flowease-tree">
  <div class="">
    <div class="simple"><span>hello</span><span>:</span><span>world</span></div>
  </div>
</div>"
`;

exports[`components > FloweaseTree > should render tree 1`] = `
"<div class="flowease-tree">
  <div class="">
    <div><span>hello</span>
      <div class="flowease-tree">
        <div class="indent">
          <div class="simple"><span>test</span><span>:</span><span>world</span></div>
        </div>
      </div>
    </div>
  </div>
  <div class="">
    <div><span>options</span>
      <div class="flowease-tree">
        <div class="indent">
          <div class="simple"><span>0</span><span>:</span><span>yes</span></div>
        </div>
        <div class="indent">
          <div class="simple"><span>1</span><span>:</span><span>no</span></div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`components > FloweaseTree > should render tree with slots 1`] = `
"<div class="flowease-tree">
  <div class="">
    <div><span>label</span>
      <div class="flowease-tree">
        <div class="indent">
          <div class="simple"><span>label</span><span>:</span><span>value</span></div>
        </div>
      </div>
    </div>
  </div>
  <div class="">
    <div><span>label</span>
      <div class="flowease-tree">
        <div class="indent">
          <div class="simple"><span>label</span><span>:</span><span>value</span></div>
        </div>
        <div class="indent">
          <div class="simple"><span>label</span><span>:</span><span>value</span></div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
